tests/integration-tests/tests/schedulers/test_slurm_accounting.py

```diff
@@ -0,0 +1,308 @@
+    results = list(
+        remote_command_executor.run_remote_script(str(test_resources_dir / "run_jobs.sh")).stdout.splitlines()
+    )
+    logging.info("Job results: %s", results)
+    for result in results:
+        logging.info("  Job Result: %s", result)
+    result = results[0].split()
+    return [
+        {
+            "node_count": len(results),
+            "account": result[1],
+            "job_id": result[2],
+            "job_name": result[3],
+            "job_nodes": [row.split()[0] for row in results],
+        }
+    ]
+
+
+def _get_connections(remote_command_executor, port):
+    connections = remote_command_executor.run_remote_command(
+        f"netstat -n --tcp | grep {port} | awk '{{print $5}}'"
+    ).stdout
+    logging.info("  Reported connections: %s", connections)
+    return [connection for connection in connections.splitlines()]
+
+
+def _test_slurmdb_users(remote_command_executor, expected_users):
+    logging.info("Testing SlurmDB Users")
+    dbusers = remote_command_executor.run_remote_command(
+        "sacctmgr list users DefaultAccount=pcdefault -Pn | cut -d '|' -f 1"
+    ).stdout
+    users = dbusers.splitlines()
+    for user in users:
+        logging.info("  User: %s", user)
+    assert_that(users).is_length(len(expected_users))
+    for user in users:
+        assert_that(user).is_in(*expected_users)
+
+
+@retry(stop_max_attempt_number=36, wait_fixed=10 * 1000)
+def _test_successful_startup_in_log(remote_command_executor):
+    log_file = "/var/log/slurmdbd.log"
+
+    log = remote_command_executor.run_remote_command("sudo cat {0}".format(log_file), hide=True).stdout
+    logging.info("slurmdbd.log: %s", log)
+    assert_that(
+        [line for line in log.splitlines() if STARTED_PATTERN.fullmatch(line) is not None], "Successful Startup"
+    ).is_not_empty()
+
+
+@retry(stop_max_attempt_number=36, wait_fixed=10 * 1000)
+def _test_slurmdbd_log_exists_in_log_group(cluster):
+    log_groups = cw_utils.get_cluster_log_groups_from_boto3(f"/aws/parallelcluster/{cluster.name}")
+    assert_that(log_groups).is_length(1)
+    log_group_name = log_groups[0].get("logGroupName")
+    log_streams = cw_utils.get_log_streams(log_group_name)
+    streams = [
+        stream.get("logStreamName")
+        for stream in log_streams
+        if re.fullmatch(r".*\.slurmdbd", stream.get("logStreamName")) is not None
+    ]
+    assert_that(streams).is_length(1)
+    stream_name = streams[0]
+    events = cw_utils.get_log_events(log_group_name, stream_name)
+    messages = (event.get("message") for event in events)
+    assert_that([message for message in messages if STARTED_PATTERN.fullmatch(message) is not None]).is_not_empty()
+
+
+def _test_getting_job_rows(remote_command_executor, expected_jobs):
+    for job in expected_jobs:
+        logging.info("Job: %s", job)
+        rows = remote_command_executor.run_remote_command(f"sacct -nP -j {job['job_id']}").stdout
```

Comment:
I believe this would be a good opportunity to extend the `slurm_commands` class with a command for retrieving job information from the database.

Reply:
Functionality moved to SlurmCommands class.